services/venice-controller/src/main/java/com/linkedin/venice/ingestion/control/MultiRegionRealTimeTopicSwitcher.java

```diff
@@ -0,0 +1,113 @@
+import com.linkedin.venice.utils.VeniceProperties;
+import com.linkedin.venice.writer.VeniceWriter;
+import com.linkedin.venice.writer.VeniceWriterFactory;
+import com.linkedin.venice.writer.VeniceWriterOptions;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class MultiRegionRealTimeTopicSwitcher extends RealTimeTopicSwitcher {
+  private static final Logger LOGGER = LogManager.getLogger(MultiRegionRealTimeTopicSwitcher.class);
+  private static final int DEFAULT_PER_DATA_CENTER_BROADCAST_TIMEOUT_IN_SECONDS = 60;
+  private final Map<String, String> allAASourceDataCenterBrokerAddressMap;
+  private final String localDataCenterName;
+
+  public MultiRegionRealTimeTopicSwitcher(
+      TopicManager topicManager,
+      VeniceWriterFactory localVeniceWriterFactory,
+      VeniceProperties veniceProperties,
+      PubSubTopicRepository pubSubTopicRepository,
+      Map<String, String> activeActiveRealTimeSourceFabricBrokerUrlMap,
+      String localDataCenterName) {
+    super(topicManager, localVeniceWriterFactory, veniceProperties, pubSubTopicRepository);
+    this.allAASourceDataCenterBrokerAddressMap = new HashMap<>(activeActiveRealTimeSourceFabricBrokerUrlMap);
+    this.localDataCenterName = localDataCenterName;
+  }
+
+  @Override
+  protected void broadcastVersionSwap(Version previousStoreVersion, Version nextStoreVersion, String topicName) {
+    String storeName = previousStoreVersion.getStoreName();
+    int partitionCount;
+    if (topicName.equals(previousStoreVersion.kafkaTopicName())) {
+      partitionCount = previousStoreVersion.getPartitionCount();
+    } else {
+      partitionCount = nextStoreVersion.getPartitionCount();
+    }
+
+    long generationId = getVersionSwapGenerationId();
+    LOGGER.info(
+        "Broadcasting Version Swap message with generation id: {}, source data center: {} to topic: {} for store: {} to {} partitions and {} data center(s)",
+        generationId,
+        localDataCenterName,
+        topicName,
+        storeName,
+        partitionCount,
+        allAASourceDataCenterBrokerAddressMap.size());
+    Map<String, CompletableFuture<Void>> dataCenterBroadcastFutureMap = new HashMap<>();
+    for (Map.Entry<String, String> entry: allAASourceDataCenterBrokerAddressMap.entrySet()) {
+      String dataCenterName = entry.getKey();
+      VeniceWriterOptions.Builder writerOptionsBuilder =
+          new VeniceWriterOptions.Builder(topicName).setTime(getTimer()).setPartitionCount(partitionCount);
+      if (!dataCenterName.equals(localDataCenterName)) {
+        writerOptionsBuilder.setBrokerAddress(entry.getValue());
+      }
+      try (VeniceWriter veniceWriter = veniceWriterFactory.createVeniceWriter(writerOptionsBuilder.build())) {
```

Comment:
I was thinking maybe we should create VWs concurrently to save some time but we can do that in a follow up PR